src/execution/expression_executor/execute_comparison.cpp

```diff
@@ -268,20 +267,31 @@ static idx_t StructSelectOperation(Vector &left, Vector &right, const SelectionV
 	D_ASSERT(lchildren.size() == rchildren.size());
 
 	for (idx_t col_no = 0; col_no < lchildren.size(); ++col_no) {
+		SelectionVector left_sel(lvdata.sel->Slice(maybe_vec, count));
```

Comment:
Does the `LIST` code need this too?

Reply:
I believe the `Compact` in the `LIST` code takes care of this for the lists. At least I could not reproduce similar issues for the lists.